packages/sdk/src/transactions/Inscriber.ts

```diff
@@ -0,0 +1,300 @@
+    if (!this.commitAddress || !this.fee) {
+      throw new Error("Invalid tx! Make sure you generate commit address or recover and finally build")
+    }
+  }
+
+  buildWitness() {
+    this.witnessScripts = {
+      inscription: buildWitnessScript({
+        mediaContent: this.mediaContent,
+        mediaType: this.mediaType,
+        meta: this.getMetadata(),
+        xkey: this.xKey
+      }),
+      recovery: buildWitnessScript({
+        mediaContent: this.mediaContent,
+        mediaType: this.mediaType,
+        meta: this.getMetadata(),
+        xkey: this.xKey,
+        recover: true
+      })
+    }
+  }
+
+  buildTaprootTree() {
+    this.buildWitness()
+    this.taprootTree = [{ output: this.witnessScripts.inscription! }, { output: this.witnessScripts.recovery! }]
+  }
+
+  getInscriptionRedeemScript(): bitcoin.payments.Payment["redeem"] {
+    return {
+      output: this.witnessScripts.inscription!,
+      redeemVersion: 192
+    }
+  }
+
+  getRecoveryRedeemScript(): bitcoin.payments.Payment["redeem"] {
+    return {
+      output: this.witnessScripts.recovery!,
+      redeemVersion: 192
+    }
+  }
+
+  private async preview({ activate }: Record<"activate", boolean> = { activate: true }) {
+    if (activate) {
+      this.previewMode = true
+      this.suitableUnspent = getDummyP2TRInput()
+      this.ready = true
+      await this.build()
+    } else {
+      this.initPSBT()
+      this.suitableUnspent = null
+      this.ready = false
+      this.inputsToSign.signingIndexes.pop() // remove last added index
+      this.outputs.pop()
+      this.previewMode = false
+    }
+  }
+
+  private restrictUsageInPreviewMode() {
+    if (this.previewMode) {
+      throw new Error("Unable to process request in preview mode")
+    }
+  }
+
+  private async calculateNetworkFeeUsingPreviewMode() {
+    await this.preview()
+    this.calculateNetworkFee()
+    await this.preview({ activate: false })
+  }
+
+  async generateCommit() {
+    this.buildTaprootTree()
+    this.payment = bitcoin.payments.p2tr({
+      internalPubkey: Buffer.from(this.xKey, "hex"),
+      network: getNetwork(this.network),
+      scriptTree: this.taprootTree,
+      redeem: this.getInscriptionRedeemScript()
+    })
+    this.witness = this.payment.witness
+
+    await this.calculateNetworkFeeUsingPreviewMode()
+
+    this.commitAddress = this.payment.address!
+    return {
+      address: this.payment.address!,
+      revealFee: this.fee + this.outputAmount
+    }
+  }
+
+  async recover() {
+    this.recovery = true
+    this.buildTaprootTree()
+
+    this.payment = createTransaction(Buffer.from(this.xKey, "hex"), "p2tr", this.network, {
+      scriptTree: this.taprootTree,
+      redeem: this.getRecoveryRedeemScript()
+    })
+
+    await this.calculateNetworkFeeUsingPreviewMode()
+  }
+
+  async isReady() {
+    this.isBuilt()
+
+    if (!this.ready) {
+      try {
+        await this.fetchAndSelectSuitableUnspent()
+      } catch (error) {
+        return false
```

Comment:
can we log the error here? because this try catch will remove all the error stack